src/app/shared/components/code-snippet/code-snippet.component.ts

```diff
@@ -0,0 +1,11 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-code-snippet',
+  templateUrl: './code-snippet.component.html',
+  styleUrls: ['./code-snippet.component.scss']
+})
+export class CodeSnippetComponent {
+  @Input() code: string | undefined;
+  @Input() title: string | null | undefined;
```

Comment:
is this type correct? for me it looks like it should be either null or undefined